### PR TITLE
Improve error handling

### PR DESCRIPTION
**Pull request by haubitzer**

### CODE REVIEW

**Review by benj56 — COMMENTED** [review]
As we will change error handling as part of the Response refactoring, I don't think we should change that now, especially since `RawApiResponse` will be removed. Therefore I'd not merge this PR unless it's really needed immediately to fix a bug

**Comment by haubitzer** [comment]
@benj56 It is needed to ensure that the error is always serializable.

**Review by benj56 — APPROVED** [review]

File: src/utils/response/rawApiResponse.ts
Comment (haubitzer) [inline src/utils/response/rawApiResponse.ts:56]: After the linter changes, this function doesn't look that nice anymore. @benj56 do you know a better way?

Comment (benj56) [reply src/utils/response/rawApiResponse.ts:56]: `in` operator to check on `unknown` instead of `any` should work

File: src/utils/response/rawApiResponse.ts
Comment (benj56) [inline src/utils/response/rawApiResponse.ts:51]: I tried adding a type, for now, that reflects what's checked in the if so that we don't have to add the eslint disables: 
```
function handleError(error: { data?: unknown; response?: { data?: unknown } }) {
  const apiResponseError: TimeTacApiError = {
    reason: ErrorReason.ReponseFailed,
    _plainError: JSON.stringify(error),
    response: undefined,
  };

  if (error.data !== undefined) {
    apiResponseError.response = error.data;
  }
  if (error.response?.data != null) {
    apiResponseError.response = error.response.data;
  }

  return Promise.reject(apiResponseError);
}
```
Maybe `{ data?: unknown} | { response?: { data?: unknown } }` would make more sense?

File: src/utils/response/rawApiResponse.ts
Comment (benj56) [inline src/utils/response/rawApiResponse.ts:67]: Here just replacing `any` with `Record<string, unknown>` works
```suggestion
function isRawApiResponse(response: Record<string, unknown>): response is RawApiResponse {
  const hasHost = 'Host' in response;
  const hasCodeversion = 'Codeversion' in response;
  const hasSuccess = 'Success' in response;
  const hasSuccessNested = 'SuccessNested' in response;
  const hasRequestStartTime = 'RequestStartTime' in response;
  const hasIgnoreTypeGuard = '_ignoreTypeGuard' in response && response._ignoreTypeGuard === true;

  return hasIgnoreTypeGuard || (hasHost && hasCodeversion && hasSuccess && hasSuccessNested && hasRequestStartTime);
}
```

File: src/utils/response/rawApiResponse.ts
Comment (haubitzer) [reply src/utils/response/rawApiResponse.ts:51]: Done with { data?: unknown} | { response?: { data?: unknown } }

**Review by benj56 — APPROVED** [review]

**Pull request merged**
